compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/BIRPackageSymbolEnter.java

```diff
@@ -1234,14 +1234,20 @@ public BType readType(int cpI) throws IOException {
                     // TODO fix
                     break;
                 case TypeTags.ERROR:
-                    BTypeSymbol errorSymbol = new BErrorTypeSymbol(SymTag.ERROR, Flags.PUBLIC, Names.EMPTY,
-                                                                   env.pkgSymbol.pkgID, null, env.pkgSymbol.owner,
-                                                                   symTable.builtinPos, COMPILED_SOURCE);
+                    pkgCpIndex = inputStream.readInt();
+                    pkgId = getPackageId(pkgCpIndex);
+                    BPackageSymbol owner = packageCache.getSymbol(pkgId);
+                    BTypeSymbol errorSymbol;
+                    if (owner != null) {
+                        errorSymbol = new BErrorTypeSymbol(SymTag.ERROR, Flags.PUBLIC, Names.EMPTY, owner.pkgID,
+                                null, owner, symTable.builtinPos, COMPILED_SOURCE);
+                    } else {
+                        errorSymbol = new BErrorTypeSymbol(SymTag.ERROR, Flags.PUBLIC, Names.EMPTY,
+                                env.pkgSymbol.pkgID, null, env.pkgSymbol, symTable.builtinPos, COMPILED_SOURCE);
```

Comment:
This change looks good. Shall we also add a comment here stating the assumption which leads to this conditional check? Basically the assumption is, if `owner` is `null`, the module this error is defined in is the current module being defined.

Reply:
@pubudu91 thanks for the suggestion. Added the comment.